core/common/src/main/java/alluxio/security/authorization/PermissionStatus.java

```diff
@@ -86,57 +99,88 @@ public FileSystemPermission getPermission() {
-   * 0777.
+   * Applies default umask to newly created files.
    *
-   * @return the default {@link PermissionStatus} for directories
+   * @param conf the runtime configuration of Alluxio
+   * @return a new {@link PermissionStatus}
+   */
+  public PermissionStatus applyFileUMask(Configuration conf) {
+    mPermission =
+        mPermission.applyUMask(FileSystemPermission.getUMask(conf)).applyUMask(FILE_UMASK);
+    return this;
+  }
+
+  /**
+   * Applies default umask to newly created directories.
+   *
+   * @param conf the runtime configuration of Alluxio
+   * @return a new {@link PermissionStatus}
    */
-  public static PermissionStatus getDirDefault() {
-    return new PermissionStatus("", "", new FileSystemPermission(Constants
-        .DEFAULT_FS_FULL_PERMISSION));
+  public PermissionStatus applyDirectoryUMask(Configuration conf) {
+    mPermission =
+        mPermission.applyUMask(FileSystemPermission.getUMask(conf));
+    return this;
   }
 
-  // TODO(binfan): remove remote parameter by making two different get methods
   /**
-   * Creates the {@link PermissionStatus} for a file or a directory.
+   * Sets the user inferred from thrift transport for this {@link PermissionStatus}. If
+   * authentication is NOSASL, this a no-op.
    *
    * @param conf the runtime configuration of Alluxio
-   * @param remote true if the request is for creating permission from client side, the
-   *               username binding into inode will be gotten from {@code AuthenticatedClientUser
-   *               .get().getName()}.
-   *               If the remote is false, the username binding into inode will be gotten from
-   *               {@link alluxio.security.LoginUser}.
    * @return the {@link PermissionStatus} for a file or a directory
    * @throws java.io.IOException when getting login user fails
    */
-  public static PermissionStatus get(Configuration conf, boolean remote) throws IOException {
+  public PermissionStatus setUserFromThriftClient(Configuration conf) throws IOException {
     if (!SecurityUtils.isAuthenticationEnabled(conf)) {
-      // no authentication
-      return new PermissionStatus("", "", FileSystemPermission.getNoneFsPermission());
+      // no authentication, no user to set
+      return this;
     }
-    if (remote) {
-      // get the username through the authentication mechanism
-      User user = AuthenticatedClientUser.get(conf);
-      if (user == null) {
-        throw new IOException(ExceptionMessage.AUTHORIZED_CLIENT_USER_IS_NULL.getMessage());
-      }
-      return new PermissionStatus(user.getName(), CommonUtils.getPrimaryGroupName(conf,
-          user.getName()), FileSystemPermission.getDefault().applyUMask(conf));
+    // get the username through the authentication mechanism
+    User user = AuthenticatedClientUser.get(conf);
+    if (user == null) {
+      throw new IOException(ExceptionMessage.AUTHORIZED_CLIENT_USER_IS_NULL.getMessage());
```

Comment:
I didn't change the exception signature but it is left from previous code.

I think both might work in this case. The `AuthenticatedClientUser.get` tries to obtain the user info from IO (here the thrift transport), it also make sense as an IO exception I guess (we haven't really get to security land in this case).

In addition, this signature change will enlarge the footprint of this PR a lot. Do you feel strong on the exception ?